project/src/main/java/com/google/sps/utilities/SpannerTasks.java

```diff
@@ -450,4 +452,44 @@ public Void run(TransactionContext transaction) throws Exception {
   private static Mutation.WriteBuilder newInsertBuilderFromOpportunitySignup() {
     return Mutation.newInsertBuilder(OPPORTUNITY_SIGNUP_TABLE);
   }
+
+  /**
+   * Given an email retrieve all events for which the user with the email is 
+   * volunteering for.
+   *
+   * @param email email for the user to retrieve events volunteering for
+   * @return events where the user is volunteering
+  */
+  public static Set<EventVolunteering> getEventsVolunteeringByEmail(String email) {
+    Set<EventVolunteering> results = new HashSet<EventVolunteering>();
+    Statement statement =
+        Statement.of(
+            String.format(
+                "SELECT Events.EventID, Events.Name, Events.Description, Events.Labels,"
+                    + " Events.Location, Events.Date, Events.Time, Events.Host,"
+                    + " VolunteeringOpportunity.Name FROM Events INNER JOIN"
+                    + " VolunteeringOpportunity ON Events.EventID ="
+                    + " VolunteeringOpportunity.EventID INNER JOIN OpportunitySignup ON"
+                    + " VolunteeringOpportunity.VolunteeringOpportunityID ="
+                    + " OpportunitySignup.VolunteeringOpportunityID WHERE Email=\"%s\"",
+                email));
+    try (ResultSet resultSet =
+        SpannerClient.getDatabaseClient().singleUse().executeQuery(statement)) {
+      while (resultSet.next()) {
+        Event event =
+            new Event.Builder(
+                    /* name = */ resultSet.getString(1),
+                    /* description = */ resultSet.getString(2),
+                    /* labels = */ new HashSet<String>(resultSet.getStringList(3)),
+                    /* location = */ resultSet.getString(4),
+                    /* date = */ resultSet.getDate(5),
+                    /* time = */ resultSet.getString(6),
+                    /* host = */ shallowReadUserFromEmail(resultSet.getString(7)).get())
```

Comment:
do we also want 
        .setOpportunities(getVolunteeringOpportunitiesByEventId(eventId))
        .setAttendees(
            shallowReadMultipleUsersFromEmails(new HashSet<String>(resultSet.getStringList(9))))
? or is it dropped for a reason

Reply:
We have chosen to only display emails in attendee list.